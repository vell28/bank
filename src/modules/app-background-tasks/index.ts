import { idleLogout } from './idleLogout';
import { backgroundUpdateAccountsBalance, backgroundUpdateAccountTask } from './accounts';

// TODO: add here loading profile, loading history, etc..
export default {
  accounts: backgroundUpdateAccountTask(),
  accountsBalance: backgroundUpdateAccountsBalance(),
  idle: idleLogout(),
};
