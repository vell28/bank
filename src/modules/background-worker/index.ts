import { forEachObjIndexed } from 'ramda';
import {
  ITask, BackgroundTask, ITaskParams, ITaskResponse, taskState
} from './task';

interface ITaskMap {
  [key: number]: ITask;
}

interface IBackgroundWorker {
  createTask: (params: ITaskParams) => ITaskResponse;
  removeTask: (id: number) => ITaskResponse;
  forceTask: (id: number) => ITaskResponse;
  removeAllTask: () => void;
}

class BackgroundWorker implements IBackgroundWorker {
  private readonly tasks: ITaskMap = {};

  public createTask(params: ITaskParams) {
    const task = new BackgroundTask(params);
    const resp = task.start();
    this.tasks[resp.id] = task;
    return resp;
  }

  public removeTask(id: number) {
    const task = this.tasks[id];
    if (!task) {
      return this.createNotFoundError(id);
    }
    delete this.tasks[id];

    return task.remove();
  }

  public forceTask(id: number) {
    const task = this.tasks[id];
    if (!task) {
      return this.createNotFoundError(id);
    }
    return task.force();
  }

  public removeAllTask() {
    forEachObjIndexed((task: ITask) => {
      task.remove();
    }, this.tasks);
  }

  // eslint-disable-next-line class-methods-use-this
  private createNotFoundError(id: number): ITaskResponse {
    return {
      state: taskState.TERMINATED,
      id,
      executeCount: -1,
    };
  }
}

export default new BackgroundWorker();
