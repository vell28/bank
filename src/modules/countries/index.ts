export const SEPA_COUNTRIES_CODES = [
  'AT',
  'BE',
  'BG',
  'HR',
  'CY',
  'CZ',
  'DK',
  'EE',
  'FI',
  'AX',
  'FR',
  'GF',
  'GP',
  'MQ',
  'YT',
  'BL',
  'MF',
  'PM',
  'DE',
  'GI',
  'GR',
  'HU',
  'IS',
  'IE',
  'IT',
  'LV',
  'LI',
  'LT',
  'LU',
  'MT',
  'MC',
  'NL',
  'NO',
  'PL',
  'PT',
  'RO',
  'SM',
  'SK',
  'SI',
  'ES',
  'SE',
  'CH',
  'GB',
];
