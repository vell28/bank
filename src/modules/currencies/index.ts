export enum CurrencyCodes {
  USD = 'USD',
  EUR = 'EUR',
  GBP = 'GBP',
  JPY = 'JPY',
  RUB = 'RUB',
  CHF = 'CHF',
  BTC = 'BTC',
  NZD = 'NZD',
  DKK = 'DKK',
  SGD = 'SGD',
  HKD = 'HKD',
  PLN = 'PLN',
  AUD = 'AUD',
  SEK = 'SEK',
  NOK = 'NOK',
  CZK = 'CZK',
  MXN = 'MXN',
  CNH = 'CNH',
}

export type CurrencyCodeType =
  | CurrencyCodes.EUR
  | CurrencyCodes.GBP
  | CurrencyCodes.JPY
  | CurrencyCodes.RUB
  | CurrencyCodes.USD
  | CurrencyCodes.CHF
  | CurrencyCodes.BTC
  | CurrencyCodes.NZD
  | CurrencyCodes.DKK
  | CurrencyCodes.SGD
  | CurrencyCodes.HKD
  | CurrencyCodes.PLN
  | CurrencyCodes.AUD
  | CurrencyCodes.SEK
  | CurrencyCodes.NOK
  | CurrencyCodes.CZK
  | CurrencyCodes.MXN
  | CurrencyCodes.CNH;

export enum CurrencyNames {
  USD = 'Dollar',
  EUR = 'Euro',
  GBP = 'Pound Sterling',
  JPY = 'Yen',
  RUB = 'Ruble',
  CHF = 'Swiss franc',
  BTC = 'Bitcoin',
  NZD = 'NZD',
  DKK = 'DKK',
  SGD = 'SGD',
  HKD = 'HKD',
  PLN = 'PLN',
  AUD = 'AUD',
  SEK = 'SEK',
  NOK = 'NOK',
  CZK = 'CZK',
  MXN = 'MXN',
  CNH = 'CNH',
}

export type CurrencyNameType =
  | CurrencyNames.USD
  | CurrencyNames.EUR
  | CurrencyNames.GBP
  | CurrencyNames.JPY
  | CurrencyNames.RUB
  | CurrencyNames.CHF
  | CurrencyNames.BTC
  | CurrencyNames.NZD
  | CurrencyNames.DKK
  | CurrencyNames.SGD
  | CurrencyNames.HKD
  | CurrencyNames.PLN
  | CurrencyNames.AUD
  | CurrencyNames.SEK
  | CurrencyNames.NOK
  | CurrencyNames.CZK
  | CurrencyNames.MXN
  | CurrencyNames.CNH;

export enum CurrencySymbols {
  USD = '$',
  EUR = '€',
  GBP = '£',
  JPY = '¥',
  RUB = '₽',
  CHF = 'CHF',
  BTC = 'BTC',
  NZD = 'NZD',
  DKK = 'DKK',
  SGD = 'SGD',
  HKD = 'HKD',
  PLN = 'PLN',
  AUD = 'AUD',
  SEK = 'SEK',
  NOK = 'NOK',
  CZK = 'CZK',
  MXN = 'MXN',
  CNH = 'CNH',
}

export type CurrencySymbolType =
  | CurrencySymbols.USD
  | CurrencySymbols.EUR
  | CurrencySymbols.GBP
  | CurrencySymbols.JPY
  | CurrencySymbols.RUB
  | CurrencySymbols.CHF
  | CurrencySymbols.BTC
  | CurrencySymbols.NZD
  | CurrencySymbols.DKK
  | CurrencySymbols.SGD
  | CurrencySymbols.HKD
  | CurrencySymbols.PLN
  | CurrencySymbols.AUD
  | CurrencySymbols.SEK
  | CurrencySymbols.NOK
  | CurrencySymbols.CZK
  | CurrencySymbols.MXN
  | CurrencySymbols.CNH;

export const CURRENCIES_IN_PROJECT: ICurrency[] = [
  {
    name: CurrencyNames.USD,
    symbol: CurrencySymbols.USD,
    code: CurrencyCodes.USD,
  },
  {
    name: CurrencyNames.EUR,
    symbol: CurrencySymbols.EUR,
    code: CurrencyCodes.EUR,
  },
  {
    name: CurrencyNames.GBP,
    symbol: CurrencySymbols.GBP,
    code: CurrencyCodes.GBP,
  },
  {
    name: CurrencyNames.JPY,
    symbol: CurrencySymbols.JPY,
    code: CurrencyCodes.JPY,
  },
  {
    name: CurrencyNames.RUB,
    symbol: CurrencySymbols.RUB,
    code: CurrencyCodes.RUB,
  },
  {
    name: CurrencyNames.CHF,
    symbol: CurrencySymbols.CHF,
    code: CurrencyCodes.CHF,
  },
  {
    name: CurrencyNames.BTC,
    symbol: CurrencySymbols.BTC,
    code: CurrencyCodes.BTC,
  },
  {
    name: CurrencyNames.NZD,
    symbol: CurrencySymbols.NZD,
    code: CurrencyCodes.NZD,
  },
  {
    name: CurrencyNames.DKK,
    symbol: CurrencySymbols.DKK,
    code: CurrencyCodes.DKK,
  },
  {
    name: CurrencyNames.SGD,
    symbol: CurrencySymbols.SGD,
    code: CurrencyCodes.SGD,
  },
  {
    name: CurrencyNames.HKD,
    symbol: CurrencySymbols.HKD,
    code: CurrencyCodes.HKD,
  },
  {
    name: CurrencyNames.PLN,
    symbol: CurrencySymbols.PLN,
    code: CurrencyCodes.PLN,
  },
  {
    name: CurrencyNames.AUD,
    symbol: CurrencySymbols.AUD,
    code: CurrencyCodes.AUD,
  },
  {
    name: CurrencyNames.SEK,
    symbol: CurrencySymbols.SEK,
    code: CurrencyCodes.SEK,
  },
  {
    name: CurrencyNames.NOK,
    symbol: CurrencySymbols.NOK,
    code: CurrencyCodes.NOK,
  },
  {
    name: CurrencyNames.CZK,
    symbol: CurrencySymbols.CZK,
    code: CurrencyCodes.CZK,
  },
  {
    name: CurrencyNames.MXN,
    symbol: CurrencySymbols.MXN,
    code: CurrencyCodes.MXN,
  },
  {
    name: CurrencyNames.CNH,
    symbol: CurrencySymbols.CNH,
    code: CurrencyCodes.CNH,
  },
];

export interface ICurrency {
  name: CurrencyNameType;
  symbol: CurrencySymbolType;
  code: CurrencyCodeType;
}

export interface ICurrencyValue {
  type: ICurrency;
  value: string;
}
