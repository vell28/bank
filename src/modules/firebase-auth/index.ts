import {
  auth, User, initializeApp, app
} from 'firebase/app';
import 'firebase/auth';
import { propOr } from 'ramda';

import { isDevEnv } from 'utils/enviroment';
import devConfig from './config/development.json';
import prodConfig from './config/production.json';

const config = isDevEnv ? devConfig : prodConfig;

const RECAPTCHA_CONTAINER_ID = 'recaptcha-container';
const RECAPTCHA_WRAPPER_ID = 'recaptcha-wrapper';

const ErrorCodes = {
  'auth/invalid-phone-number': 'Invalid phone number, please try again',
  'auth/invalid-verification-code': 'The SMS verification code used to create the phone auth credential is invalid.',
};

interface IRespError {
  code: string;
  message: string;
}

export class AuthFirebaseService {
  private recaptchaVerified?: auth.RecaptchaVerifier;

  private confirmationResult?: auth.ConfirmationResult;

  private firebase: app.App;

  constructor(options: object) {
    this.firebase = initializeApp(options);
    /*   if (isDevEnv) {
      this.firebase.auth().settings.appVerificationDisabledForTesting = true;
    } */
  }

  public getLoginCode(phoneNumber: string): Promise<string> {
    this.resetReCaptcha();

    this.recaptchaVerified = new auth.RecaptchaVerifier(RECAPTCHA_CONTAINER_ID, {
      size: 'invisible',
    });

    return this.firebase
      .auth()
      .signInWithPhoneNumber(phoneNumber, this.recaptchaVerified)
      .then((result: auth.ConfirmationResult) => {
        this.confirmationResult = result;
        return '';
      })
      .catch((e: IRespError) => {
        const message = propOr('SMS not sent', 'message')(e);
        return Promise.reject(propOr(message, e.code)(ErrorCodes));
      });
  }

  public async verifyCode(verificationCode: string): Promise<string> {
    if (!this.confirmationResult) {
      // eslint-disable-next-line prefer-promise-reject-errors
      return Promise.reject('Missing auth confirmation result');
    }

    return this.confirmationResult
      .confirm(verificationCode)
      .then((userCredential: auth.UserCredential) => {
        return (userCredential.user && userCredential.user.getIdToken()) || '';
      })
      .catch((e: IRespError) => {
        const message = propOr('Invalid confirm code', 'message')(e);
        return Promise.reject(propOr(message, e.code)(ErrorCodes));
      });
  }

  public async refreshToken() {
    const currentUser = this.getAuthState();
    if (currentUser) {
      return currentUser
        .getIdToken(true)
        .then((token: string) => {
          return token;
        })
        .catch((e: IRespError) => {
          const message = propOr('Can`t refresh firebase token', 'message')(e);
          return Promise.reject(propOr(message, e.code)(ErrorCodes));
        });
    }
    // eslint-disable-next-line prefer-promise-reject-errors
    return Promise.reject('Can`t find firebase user');
  }

  public getAuthState(): User | null {
    return this.firebase.auth().currentUser;
  }

  public logout(): void {
    this.resetReCaptcha();
    this.firebase.auth().signOut();
  }

  public resetReCaptcha() {
    if (this.recaptchaVerified) {
      this.recaptchaVerified.clear();
      const wrapper = document.getElementById(RECAPTCHA_WRAPPER_ID);
      if (wrapper) {
        wrapper.innerHTML = `<div id="${RECAPTCHA_CONTAINER_ID}" style="display:none;"></div>`;
      }
    }
  }
}

export const firebaseAuth = new AuthFirebaseService(config);
