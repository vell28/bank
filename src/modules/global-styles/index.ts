import { createGlobalStyle } from 'styled-components';

// import { fontStyle } from './fonts';
import './css/swiper.css';
import '../../../node_modules/@fortawesome/fontawesome-free/css/all.css';
// import 'flatpickr/dist/themes/material_green.css';
import './css/flatpickr.css';

import { typographyStyles } from './typography';

export const GlobalStyle = createGlobalStyle`
  /* typography styles */
  ${typographyStyles}
  body {}
  #root {
    min-height: 100%;
    height: auto;
  }
  #static {
    display: none;
  }
  input {
     ::placeholder { color: ${(props) => props.theme.colors.black}20; font-weight: normal; }
     :-ms-input-placeholder { color: ${(props) => props.theme.colors.black}20; font-weight: normal; }
     ::-ms-input-placeholder { color: ${(props) => props.theme.colors.black}20; font-weight: normal; }
  }
  select {
    border-radius: 0;
  }
  select {
    border-radius: 0;
  }
`;
