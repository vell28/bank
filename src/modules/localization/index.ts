// @flow

import i18n from 'i18next';
import { find, test } from 'ramda';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

// ToDo: need sync to existed locales
import { de, en } from './locales';
import { ILanguage, LangKeysType, LanguageKeys } from './entities';

export const languagesList: ILanguage[] = [
  { key: LanguageKeys.EN, nativeName: 'English' },
  { key: LanguageKeys.DE, nativeName: 'Deutsch' },
];

export const languagesMap = {
  [LanguageKeys.DE]: 'Deutsch',
  [LanguageKeys.EN]: 'English',
  [LanguageKeys.RU]: 'Russian',
};

export const getActiveLanguageKey = (activeLanguage: string = i18n.language): LangKeysType => {
  // When localStorage is empty, activeLanguage='en-GB'
  const item = find((lang: ILanguage) => test(new RegExp(lang.key, 'gi'), activeLanguage), languagesList);
  if (item) {
    return item.key;
  }
  return LanguageKeys.EN;
};

export const changeLanguage = (lng: LangKeysType) => {
  window.location.reload();
  i18n.changeLanguage(lng);
};

/**
 * Detector Options
 * https://github.com/i18next/i18next-browser-languageDetector#detector-options
 */
const detection = {
  caches: ['localStorage'],
};

/**
 * ISO 639
 * https://www.iso.org/iso-639-language-codes.html
 * http://www.loc.gov/standards/iso639-2/php/code_list.php
 */
i18n
  // .use(Locize)
  .use(initReactI18next) // passes i18n down to react-i18next
  .use(LanguageDetector)
  .init({
    ns: ['common'],
    defaultNS: 'common',

    resources: { en, de },

    fallbackLng: LanguageKeys.EN,

    nsSeparator: '::',
    keySeparator: false,

    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },

    detection,
  });

export default i18n;
