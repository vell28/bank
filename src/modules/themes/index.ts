import { IThemeColors, colorPalette } from './colorPalettes';
import {
  SMALL_VIEWPORT, MEDIUM_VIEWPORT, LARGE_VIEWPORT, MOBILE_VIWEPORT, fonts
} from './config';

export interface ITheme {
  name: string;
  colors: IThemeColors;
  dimension: object;
  size: string;
  breakpoints: string[];
  fonts: {
    faFontsPath: string;
    fontFamily: string;
  };
  typography: {
    bodyBackground: string;
    bodyFontColor: string;
    bodyFontFamily: string;

    globalFontSize: string;
    globalWeightNormal: number;
    globalLineHeight: number;

    headersFontColor: string;
    headersFontFamily: string;
    headersFontWeight: number;
  };
}

export const mainDarkTheme: ITheme = {
  name: 'main-dark-theme',
  colors: colorPalette,
  dimension: {},
  size: 'default',
  breakpoints: [`${MOBILE_VIWEPORT}px`, `${SMALL_VIEWPORT}px`, `${MEDIUM_VIEWPORT}px`, `${LARGE_VIEWPORT}px`],
  fonts,
  typography: {
    bodyBackground: colorPalette.darkBlue,
    bodyFontColor: colorPalette.white,
    bodyFontFamily: fonts.fontFamily,

    globalFontSize: '1.6rem',
    globalWeightNormal: 400,
    globalLineHeight: 1.5,

    headersFontColor: colorPalette.white,
    headersFontFamily: fonts.fontFamily,
    headersFontWeight: 700,
  },
};
