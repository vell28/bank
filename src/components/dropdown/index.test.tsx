import React from 'react';

import { ThemeProvider } from 'styled-components';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { mainDarkTheme } from 'modules/themes';
import { SidebarDropDown } from '.';

const InnerRender = () => (
  <div className="inner-render">
    Account
    {' '}
    <br />
    {' '}
    line1
    {' '}
    <br />
    {' '}
    line2
  </div>
);

describe('<SidebarDropDown /> tests', () => {
  test('should success render, should open/close inner func', () => {
    const component: any = mount(
      <ThemeProvider theme={mainDarkTheme}>
        <SidebarDropDown render={<InnerRender />}>
          <div className="open-btn">openBtn</div>
        </SidebarDropDown>
      </ThemeProvider>,
    );
    /*    const btnOpen = component.find('.open-btn');
    const emptyMenu = component.find('.inner-render');
    expect(emptyMenu.length).toEqual(0);

    btnOpen.simulate('click');

    const notEmptyMenu = component.find('.inner-render');

    expect(notEmptyMenu.length).toEqual(1); */
    // TODO: add checking overflow hidden
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should been closed with external handle', () => {
    const component: any = mount(
      <ThemeProvider theme={mainDarkTheme}>
        <SidebarDropDown render={<InnerRender />} isShown={false}>
          <div className="open-btn">openBtn</div>
        </SidebarDropDown>
      </ThemeProvider>,
    );
    // const emptyMenu = component.find('.inner-render');
    // expect(emptyMenu.length).toEqual(0);
    // TODO: add checking overflow hidden
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should been opened with external handle', () => {
    const component: any = mount(
      <ThemeProvider theme={mainDarkTheme}>
        <SidebarDropDown render={<InnerRender />} isShown>
          <div className="open-btn">openBtn</div>
        </SidebarDropDown>
      </ThemeProvider>,
    );
    const emptyMenu = component.find('.inner-render');
    expect(emptyMenu.length).toEqual(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
