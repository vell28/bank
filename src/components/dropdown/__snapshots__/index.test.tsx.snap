// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarDropDown /> tests should been closed with external handle 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Array [
        "576px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "alto": "#dedede",
        "asparagus": "#77a45c",
        "black": "#000000",
        "blue": "#008cff",
        "blueLight": "#70cde4",
        "borderLight": "#cad3db",
        "dark": "#030303",
        "darkBlue": "#011f3d",
        "darkBlue2": "#132F4B",
        "elephant": "#12304b",
        "feijoa": "#93d77e",
        "gray": "#a5adb8",
        "gray10": "#cad3db",
        "gray11": "#959fac",
        "gray2": "#a7a9ac",
        "gray3": "#a1aab6",
        "gray4": "#e8e8e8",
        "gray5": "#cbcbcb",
        "gray6": "#eaeaea",
        "gray7": "#acacac",
        "gray9": "#9faab7",
        "grayHover": "#00000008",
        "grayLight": "#cdd0d5",
        "green": "#7ed321",
        "green1": "#5ba00f",
        "green2": "#4dd964",
        "green3": "#77a45c",
        "green4": "#b1de6b",
        "greenLight": "#d9f9b6",
        "light": "#d5dae0",
        "malibu": "#53b8ff",
        "navy": "#01172d",
        "orange": "#e4be79",
        "persimmon": "#ff5353",
        "purple": "#a479cb",
        "red": "#e10606",
        "red2": "#ff2e7b",
        "red3": "#ea4848",
        "rose": "#d08484",
        "slate": "#596778",
        "submarine": "#b6bcc4",
        "turquoise": "#79cbcb",
        "white": "#ffffff",
        "white2": "#dddddd",
      },
      "dimension": Object {},
      "fonts": Object {
        "faFontsPath": "/css/webfonts",
        "fontFamily": "AvenirNext, Arial",
      },
      "name": "main-dark-theme",
      "size": "default",
      "typography": Object {
        "bodyBackground": "#011f3d",
        "bodyFontColor": "#ffffff",
        "bodyFontFamily": "AvenirNext, Arial",
        "globalFontSize": "1.6rem",
        "globalLineHeight": 1.5,
        "globalWeightNormal": 400,
        "headersFontColor": "#ffffff",
        "headersFontFamily": "AvenirNext, Arial",
        "headersFontWeight": 700,
      },
    }
  }
>
  <SidebarDropDown
    isShown={false}
    render={<InnerRender />}
  >
    <styled.div>
      <div
        className="sc-AxjAm hZFCCY"
      >
        <styled.div
          onClick={[Function]}
        >
          <div
            className="sc-AxirZ gpWHsK"
            onClick={[Function]}
          >
            <div
              className="open-btn"
            >
              openBtn
            </div>
          </div>
        </styled.div>
        <styled.div
          className=""
          radius={25}
        >
          <div
            className="sc-AxhCb cbKhiq"
            radius={25}
          >
            <styled.div
              radius={25}
            >
              <div
                className="sc-AxiKw cEnCFp"
                radius={25}
              >
                <InnerRender>
                  <div
                    className="inner-render"
                  >
                    Account
                     
                    <br />
                     
                    line1
                     
                    <br />
                     
                    line2
                  </div>
                </InnerRender>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </SidebarDropDown>
</ThemeProvider>
`;

exports[`<SidebarDropDown /> tests should been opened with external handle 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Array [
        "576px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "alto": "#dedede",
        "asparagus": "#77a45c",
        "black": "#000000",
        "blue": "#008cff",
        "blueLight": "#70cde4",
        "borderLight": "#cad3db",
        "dark": "#030303",
        "darkBlue": "#011f3d",
        "darkBlue2": "#132F4B",
        "elephant": "#12304b",
        "feijoa": "#93d77e",
        "gray": "#a5adb8",
        "gray10": "#cad3db",
        "gray11": "#959fac",
        "gray2": "#a7a9ac",
        "gray3": "#a1aab6",
        "gray4": "#e8e8e8",
        "gray5": "#cbcbcb",
        "gray6": "#eaeaea",
        "gray7": "#acacac",
        "gray9": "#9faab7",
        "grayHover": "#00000008",
        "grayLight": "#cdd0d5",
        "green": "#7ed321",
        "green1": "#5ba00f",
        "green2": "#4dd964",
        "green3": "#77a45c",
        "green4": "#b1de6b",
        "greenLight": "#d9f9b6",
        "light": "#d5dae0",
        "malibu": "#53b8ff",
        "navy": "#01172d",
        "orange": "#e4be79",
        "persimmon": "#ff5353",
        "purple": "#a479cb",
        "red": "#e10606",
        "red2": "#ff2e7b",
        "red3": "#ea4848",
        "rose": "#d08484",
        "slate": "#596778",
        "submarine": "#b6bcc4",
        "turquoise": "#79cbcb",
        "white": "#ffffff",
        "white2": "#dddddd",
      },
      "dimension": Object {},
      "fonts": Object {
        "faFontsPath": "/css/webfonts",
        "fontFamily": "AvenirNext, Arial",
      },
      "name": "main-dark-theme",
      "size": "default",
      "typography": Object {
        "bodyBackground": "#011f3d",
        "bodyFontColor": "#ffffff",
        "bodyFontFamily": "AvenirNext, Arial",
        "globalFontSize": "1.6rem",
        "globalLineHeight": 1.5,
        "globalWeightNormal": 400,
        "headersFontColor": "#ffffff",
        "headersFontFamily": "AvenirNext, Arial",
        "headersFontWeight": 700,
      },
    }
  }
>
  <SidebarDropDown
    isShown={true}
    render={<InnerRender />}
  >
    <styled.div>
      <div
        className="sc-AxjAm hZFCCY"
      >
        <styled.div
          onClick={[Function]}
        >
          <div
            className="sc-AxirZ gpWHsK"
            onClick={[Function]}
          >
            <div
              className="open-btn"
            >
              openBtn
            </div>
          </div>
        </styled.div>
        <styled.div
          className="shown"
          radius={25}
        >
          <div
            className="sc-AxhCb cbKhiq shown"
            radius={25}
          >
            <styled.div
              radius={25}
            >
              <div
                className="sc-AxiKw cEnCFp"
                radius={25}
              >
                <InnerRender>
                  <div
                    className="inner-render"
                  >
                    Account
                     
                    <br />
                     
                    line1
                     
                    <br />
                     
                    line2
                  </div>
                </InnerRender>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </SidebarDropDown>
</ThemeProvider>
`;

exports[`<SidebarDropDown /> tests should success render, should open/close inner func 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Array [
        "576px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "alto": "#dedede",
        "asparagus": "#77a45c",
        "black": "#000000",
        "blue": "#008cff",
        "blueLight": "#70cde4",
        "borderLight": "#cad3db",
        "dark": "#030303",
        "darkBlue": "#011f3d",
        "darkBlue2": "#132F4B",
        "elephant": "#12304b",
        "feijoa": "#93d77e",
        "gray": "#a5adb8",
        "gray10": "#cad3db",
        "gray11": "#959fac",
        "gray2": "#a7a9ac",
        "gray3": "#a1aab6",
        "gray4": "#e8e8e8",
        "gray5": "#cbcbcb",
        "gray6": "#eaeaea",
        "gray7": "#acacac",
        "gray9": "#9faab7",
        "grayHover": "#00000008",
        "grayLight": "#cdd0d5",
        "green": "#7ed321",
        "green1": "#5ba00f",
        "green2": "#4dd964",
        "green3": "#77a45c",
        "green4": "#b1de6b",
        "greenLight": "#d9f9b6",
        "light": "#d5dae0",
        "malibu": "#53b8ff",
        "navy": "#01172d",
        "orange": "#e4be79",
        "persimmon": "#ff5353",
        "purple": "#a479cb",
        "red": "#e10606",
        "red2": "#ff2e7b",
        "red3": "#ea4848",
        "rose": "#d08484",
        "slate": "#596778",
        "submarine": "#b6bcc4",
        "turquoise": "#79cbcb",
        "white": "#ffffff",
        "white2": "#dddddd",
      },
      "dimension": Object {},
      "fonts": Object {
        "faFontsPath": "/css/webfonts",
        "fontFamily": "AvenirNext, Arial",
      },
      "name": "main-dark-theme",
      "size": "default",
      "typography": Object {
        "bodyBackground": "#011f3d",
        "bodyFontColor": "#ffffff",
        "bodyFontFamily": "AvenirNext, Arial",
        "globalFontSize": "1.6rem",
        "globalLineHeight": 1.5,
        "globalWeightNormal": 400,
        "headersFontColor": "#ffffff",
        "headersFontFamily": "AvenirNext, Arial",
        "headersFontWeight": 700,
      },
    }
  }
>
  <SidebarDropDown
    render={<InnerRender />}
  >
    <styled.div>
      <div
        className="sc-AxjAm hZFCCY"
      >
        <styled.div
          onClick={[Function]}
        >
          <div
            className="sc-AxirZ gpWHsK"
            onClick={[Function]}
          >
            <div
              className="open-btn"
            >
              openBtn
            </div>
          </div>
        </styled.div>
        <styled.div
          className=""
          radius={25}
        >
          <div
            className="sc-AxhCb cbKhiq"
            radius={25}
          >
            <styled.div
              radius={25}
            >
              <div
                className="sc-AxiKw cEnCFp"
                radius={25}
              >
                <InnerRender>
                  <div
                    className="inner-render"
                  >
                    Account
                     
                    <br />
                     
                    line1
                     
                    <br />
                     
                    line2
                  </div>
                </InnerRender>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </SidebarDropDown>
</ThemeProvider>
`;
